src/pkg/cli/client/byoc/gcp/byoc.go

```diff
@@ -0,0 +1,623 @@
+package gcp
+
+import (
+	"bytes"
+	"context"
+	"encoding/base64"
+	"errors"
+	"fmt"
+	"os"
+	"path"
+	"strings"
+	"time"
+
+	"cloud.google.com/go/storage"
+	"github.com/DefangLabs/defang/src/pkg"
+	"github.com/DefangLabs/defang/src/pkg/cli/client"
+	"github.com/DefangLabs/defang/src/pkg/cli/client/byoc"
+	"github.com/DefangLabs/defang/src/pkg/cli/compose"
+	"github.com/DefangLabs/defang/src/pkg/clouds/aws/ecs"
+	"github.com/DefangLabs/defang/src/pkg/clouds/gcp"
+	"github.com/DefangLabs/defang/src/pkg/http"
+	"github.com/DefangLabs/defang/src/pkg/term"
+	"github.com/DefangLabs/defang/src/pkg/types"
+	defangv1 "github.com/DefangLabs/defang/src/protos/io/defang/v1"
+	"github.com/aws/smithy-go/ptr"
+	"github.com/bufbuild/connect-go"
+	composeTypes "github.com/compose-spec/compose-go/v2/types"
+	"google.golang.org/api/googleapi"
+	"google.golang.org/protobuf/proto"
+)
+
+var _ client.Provider = (*ByocGcp)(nil)
+
+const (
+	DefangCDProjectName = "defang-cd"
+	UploadPrefix        = "uploads/"
+)
+
+var (
+	DefaultCDTags    = map[string]string{"created-by": "defang"}
+	PulumiVersion    = pkg.Getenv("DEFANG_PULUMI_VERSION", "3.136.1")
+	DefangGcpCdImage = pkg.Getenv("DEFANG_GCP_CD_IMAGE", "edwardrf/gcpcd:test")
```

Comment:
This probably needs to get replaced before merging

Reply:
Yes, done, pending fabric deployment.